File: langgraph/data/equiv/21.py
from typing import List


def normalize_values(values: List[float]) -> List[float]:
    smallest_value = min(values)
    largest_value = max(values)
    range_value = largest_value - smallest_value

    normalized_list = []
    for value in values:
        normalized_value = (value - smallest_value) / range_value
        normalized_list.append(normalized_value)

    return normalized_list
# method name: rescale_to_unit -> normalize_values
# parameter name: numbers -> values
# variable name: min_number -> smallest_value, max_number -> largest_value
# introduced range_value to store the difference between largest and smallest values
# replaced list comprehension with a for loop to append each normalized value

File: langgraph/result/qwen/32b-eq2js/21.js
function normalizeValues(values) {
    const smallestValue = Math.min(...values);
    const largestValue = Math.max(...values);
    const rangeValue = largestValue - smallestValue;

    const normalizedList = [];
    for (const value of values) {
        const normalizedValue = (value - smallestValue) / rangeValue;
        normalizedList.push(normalizedValue);
    }

    return normalizedList;
}